### PR TITLE
initial samples notifications

### PR DESCRIPTION
**Pull request by rafaelMurata**
## Description

Fixes #<ISSUE-NUMBER>

Note: Before submitting a pull request, please open an issue for discussion if you are not associated with Google.

## Checklist

- [x] I have followed [Sample Format Guide](https://github.com/GoogleCloudPlatform/java-docs-samples/blob/main/SAMPLE_FORMAT.md)
- [ ] `pom.xml` parent set to latest `shared-configuration`
- [ ] Appropriate changes to README are included in PR
- [ ] These samples need a new **API enabled** in testing projects to pass (let us know which ones)
- [ ] These samples need a new/updated **env vars** in testing projects set to pass (let us know which ones)
- [x] **Tests** pass:   `mvn clean verify` **required**
- [x] **Lint**  passes: `mvn -P lint checkstyle:check` **required**
- [ ] **Static Analysis**:  `mvn -P lint clean compile pmd:cpd-check spotbugs:check` **advisory only**
- [ ] This sample adds a new sample directory, and I updated the [CODEOWNERS file](https://github.com/GoogleCloudPlatform/java-docs-samples/blob/main/.github/CODEOWNERS) with the codeowners for this sample
- [ ] This sample adds a new **Product API**, and I updated the [Blunderbuss issue/PR auto-assigner](https://github.com/GoogleCloudPlatform/java-docs-samples/blob/main/.github/blunderbuss.yml) with the codeowners for this sample 
- [x] Please **merge** this PR for me once it is approved

### CODE REVIEW

File: security-command-center/snippets/src/main/java/vtwo/notifications/CreateNotification.java [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+import com.google.cloud.securitycenter.v2.SecurityCenterClient;
+import java.io.IOException;
+
+public class CreateNotification {
+    public static void main(String[] args) throws IOException {
+        // parentId: must be in one of the following formats:
+        //    "organizations/{organization_id}"
+        //    "projects/{project_id}"
+        //    "folders/{folder_id}"
+        String projectId = "{project-id}";
```
Comment (arthurlapertosa) [inline security-command-center/snippets/src/main/java/vtwo/notifications/CreateNotification.java]: Given that the comment above says `parentId`, don't you think it's better for this line to be:
```
String parentId = "{parentId}";
```

File: security-command-center/snippets/src/main/java/vtwo/notifications/CreateNotification.java [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+        String topicName = "{your-topic}";
+        String notificationConfigId = "{your-notification-id}";
+        // Specify the location of the notification config.
+        String location = "global";
+
+        createNotificationConfig(projectId, location , topicName, notificationConfigId);
+    }
+
+    // Crete a notification config.
```
Comment (arthurlapertosa) [inline security-command-center/snippets/src/main/java/vtwo/notifications/CreateNotification.java]: Typo: Crete -> Create

File: security-command-center/snippets/src/main/java/vtwo/notifications/DeleteNotification.java [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+import java.io.IOException;
+
+public class DeleteNotification {
+    public static void main(String[] args) throws IOException {
+        // parentId: must be in one of the following formats:
+        //    "organizations/{organization_id}"
+        //    "projects/{project_id}"
+        //    "folders/{folder_id}"
+        String projectId = "{project-id}";
```
Comment (arthurlapertosa) [inline security-command-center/snippets/src/main/java/vtwo/notifications/DeleteNotification.java]: Same as the other file

File: security-command-center/snippets/src/main/java/vtwo/notifications/DeleteNotification.java [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+        // Specify the location to list the findings.
+        String location = "global";
+        String notificationConfigId = "{your-notification-id}";
+
+        deleteNotificationConfig(projectId, location,notificationConfigId);
+    }
+    // Delete a notification config.
```
Comment (arthurlapertosa) [inline security-command-center/snippets/src/main/java/vtwo/notifications/DeleteNotification.java]: add space above (although I think the space will be auto added once you install the java formatter)

File: security-command-center/snippets/src/main/java/vtwo/notifications/GetNotification.java [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+import java.io.IOException;
+
+public class GetNotification {
+    public static void main(String[] args) throws IOException {
+        // parentId: must be in one of the following formats:
+        //    "organizations/{organization_id}"
+        //    "projects/{project_id}"
+        //    "folders/{folder_id}"
+
+        String projectId = "{your-project}";
```
Comment (arthurlapertosa) [inline security-command-center/snippets/src/main/java/vtwo/notifications/GetNotification.java]: same here

File: security-command-center/snippets/src/main/java/vtwo/notifications/GetNotification.java [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+        // once, and can be reused for multiple requests. After completing all of your requests, call
+        // the "close" method on the client to safely clean up any remaining background resources.
+        try (SecurityCenterClient client = SecurityCenterClient.create()) {
+            GetNotificationConfigRequest request = GetNotificationConfigRequest.newBuilder()
+                    .setName(String.format("projects/%s/locations/%s/notificationConfigs/%s",
+                            projectId,
+                            location,
+                            notificationConfigId))
+                    .build();
+            System.out.printf("getNotificationConfig config::" +String.format("projects/%s/locations/%s/notificationConfigs/%s",
```
Comment (arthurlapertosa) [inline security-command-center/snippets/src/main/java/vtwo/notifications/GetNotification.java]: extra `:`
`getNotificationConfig config::` -> `getNotificationConfig config:`

File: security-command-center/snippets/src/main/java/vtwo/notifications/ListNotification.java [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+import java.io.IOException;
+
+public class ListNotification {
+    public static void main(String[] args) throws IOException {
+        // parentId: must be in one of the following formats:
+        //    "organizations/{organization_id}"
+        //    "projects/{project_id}"
+        //    "folders/{folder_id}"
+
+        String projectId = "{your-project}";
```
Comment (arthurlapertosa) [inline security-command-center/snippets/src/main/java/vtwo/notifications/ListNotification.java]: same here

File: security-command-center/snippets/src/main/java/vtwo/notifications/ListNotification.java [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+        listNotificationConfigs(projectId,location);
+    }
+
+    // List notification configs present in the given parent.
+    public static ImmutableList<NotificationConfig> listNotificationConfigs(String projectId, String location)
+            throws IOException {
+        // Initialize client that will be used to send requests. This client only needs to be created
+        // once, and can be reused for multiple requests. After completing all of your requests, call
+        // the "close" method on the client to safely clean up any remaining background resources.
+
```
Comment (arthurlapertosa) [inline security-command-center/snippets/src/main/java/vtwo/notifications/ListNotification.java:41]: extra space here

File: security-command-center/snippets/src/main/java/vtwo/notifications/GetNotification.java [hunk 9]
```diff
@@ -0,0 +1,68 @@
...
+
+        getNotificationConfig(projectId, location,notificationConfigId);
+    }
+
+    // Retrieve an existing notification config.
+    public static NotificationConfig getNotificationConfig(
+            String projectId,String location ,String notificationConfigId) throws IOException {
+        // Initialize client that will be used to send requests. This client only needs to be created
+        // once, and can be reused for multiple requests. After completing all of your requests, call
+        // the "close" method on the client to safely clean up any remaining background resources.
```
Comment (arthurlapertosa) [inline security-command-center/snippets/src/main/java/vtwo/notifications/GetNotification.java]: Although it says here that you should call the "close" method, you never do throughout the code. Is there a particular reason why?

File: security-command-center/snippets/src/main/java/vtwo/notifications/UpdateNotification.java [hunk 10]
```diff
@@ -0,0 +1,86 @@
...
+import java.io.IOException;
+import java.util.UUID;
+public class UpdateNotification {
+
+    public static void main(String[] args) throws IOException {
+        // parentId: must be in one of the following formats:
+        //    "organizations/{organization_id}"
+        //    "projects/{project_id}"
+        //    "folders/{folder_id}"
+        String projectId = "{project-id}";
```
Comment (arthurlapertosa) [inline security-command-center/snippets/src/main/java/vtwo/notifications/UpdateNotification.java]: same here

File: security-command-center/snippets/src/test/java/vtwo/NotificationIT.java [hunk 11]
```diff
@@ -0,0 +1,113 @@
...
+    public final MultipleAttemptsRule multipleAttemptsRule = new MultipleAttemptsRule(
+            MAX_ATTEMPT_COUNT,
+            INITIAL_BACKOFF_MILLIS);
+
+    // Check if the required environment variables are set.
+    public static void requireEnvVar(String envVarName) {
+        assertWithMessage(String.format("Missing environment variable '%s' ", envVarName))
+                .that(System.getenv(envVarName))
+                .isNotEmpty();
+    }
```
Comment (arthurlapertosa) [inline security-command-center/snippets/src/test/java/vtwo/NotificationIT.java]: add space below (also add a space after each method below)

File: security-command-center/snippets/src/main/java/vtwo/notifications/GetNotification.java [hunk 12]
```diff
@@ -0,0 +1,68 @@
...
+
+import java.io.IOException;
+
+public class GetNotification {
+    public static void main(String[] args) throws IOException {
+        // parentId: must be in one of the following formats:
+        //    "organizations/{organization_id}"
+        //    "projects/{project_id}"
+        //    "folders/{folder_id}"
+
```
Comment (arthurlapertosa) [inline security-command-center/snippets/src/main/java/vtwo/notifications/GetNotification.java:27]: delete extra space

File: security-command-center/snippets/src/main/java/vtwo/notifications/ListNotification.java [hunk 13]
```diff
@@ -0,0 +1,68 @@
...
+import java.io.IOException;
+
+public class ListNotification {
+    public static void main(String[] args) throws IOException {
+        // parentId: must be in one of the following formats:
+        //    "organizations/{organization_id}"
+        //    "projects/{project_id}"
+        //    "folders/{folder_id}"
+
```
Comment (arthurlapertosa) [inline security-command-center/snippets/src/main/java/vtwo/notifications/ListNotification.java]: delete extra space

File: security-command-center/snippets/src/main/java/vtwo/notifications/ListNotification.java [hunk 14]
```diff
@@ -0,0 +1,68 @@
...
+        String projectId = "{your-project}";
+
```
Comment (arthurlapertosa) [inline security-command-center/snippets/src/main/java/vtwo/notifications/ListNotification.java:29]: delete extra space

**Review by arthurlapertosa — CHANGES_REQUESTED** [review]

File: security-command-center/snippets/src/main/java/vtwo/notifications/GetNotification.java (hunk 9 above)
Comment (rafaelMurata) [reply security-command-center/snippets/src/main/java/vtwo/notifications/GetNotification.java]: I just followed the example of the GetNotificationConfigSnippets class

**Comment by arthurlapertosa** [comment]
please close this PR

**Pull request closed without merging**